src/design/design.controller.ts

```diff
@@ -51,6 +51,17 @@ export class DesignController {
     return this.service.create(dto);
   }
 
+  @ApiOperation({ summary: 'Create a new design (full)' })
+  @ApiResponse({
+    status: 201,
+  })
+  @Post('/designs/full')
+  @UseGuards(JwtAuthGuard)
+  async createFull(@Request() req, @Body() dto: any): Promise<any> {
```

Comment:
Can we use a typed DTO here with validation?
Can we type the return value?

Reply:
This is just a passthrough, we have no entities, types or dto-s related to designs here. Basically all validation, dto typing etc. is happening on services side, and all validation errors (and other errors) are forwarded to the Ablo demo FE. That's why there are no tests too. We could duplicate those things on ablo-demo too if needed, but then we're mocking the real design endpoint so idk if the tests make sense

Reply:
> This is just a passthrough, we have no entities, types or dto-s related to designs here. Basically all validation, dto typing etc. is happening on services side, and all validation errors (and other errors) are forwarded to the Ablo demo FE. That's why there are no tests too. We could duplicate those things on ablo-demo too if needed, but then we're mocking the real design endpoint so idk if the tests make sense

That's fair. It just seems kind of sloppy given we open sourced this and it could one day be the BE for our own B2C product. Let me think about it some more, but we can push this through